TypeCobol/Analysis/AnalyzerProvider.cs

```diff
@@ -70,14 +70,22 @@ private TAnalyzer[] Concat<TAnalyzer>(TAnalyzer[] fromBase, Func<IAnalyzerProvid
         {
             if (_providers != null && _providers.Count > 0)
             {
-                var result = _providers
-                    .Select(selector) //from provider to analyzers
-                    .Where(a => a != null) //providers may return null arrays so filter them out
-                    .SelectMany(a => a); //join all arrays into a single IEnumerable
-                if (fromBase != null)
+                var result = new List<TAnalyzer>();
+                if (fromBase != null) result.AddRange(fromBase);
+                foreach (var analyzerProvider in _providers)
                 {
-                    result = fromBase.Concat(result);
+                    try
+                    {
+                        var analyzers = selector(analyzerProvider);
+                        result.AddRange(analyzers);
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e);
+                        throw;
+                    }
                 }
+                
```

Comment:
In fact, we don't have to change the `CompositeAnalyzerProvider`, we must secure the code at 3 locations:
- Load
- Create
- Call to analyzer itself

For load operations, it's ok with what you've done: the unsafe load is already try-catch protected and the `AddCustomProviders` is now safe.
For call to analyzer we are missing the protection on the call to `GetResult` for syntax-driven analyzers
For the create part, you secured the quality analyzer case in CompilationUnit.cs but we need to do the same for syntax-driven. So I would like you to reset `CompositeAnalyzerProvider` to its original state and add 2 try-catch blocks in `ProduceTemporarySemanticDocument` to secure the creation and result collection phases.